optimade/models/tests/test_models.py

```diff
@@ -1,7 +1,49 @@
+from pathlib import Path
+
 import pytest
-from pydantic import BaseModel, ValidationError, ConfigError
+import unittest
+import json
+import bson.json_util
+
+from pydantic import ValidationError, BaseModel, ConfigError
+from optimade.models.util import conlist, ManualValidationError
+from optimade.models import StructureResource
+from optimade.server.mappers import StructureMapper
+
+
+class TestPydanticValidation(unittest.TestCase):
+    def test_good_structures(self):
+        test_structures_path = (
+            Path(__file__)
+            .resolve()
+            .parent.joinpath("../../server/tests/test_structures.json")
+        )
+        with open(test_structures_path, "r") as f:
+            good_structures = json.load(f)
+            # adjust dates as mongo would
+        for doc in good_structures:
+            doc["last_modified"] = doc["last_modified"]["$date"]
+
+        for structure in good_structures:
+            _ = StructureResource(**StructureMapper.map_back(structure))
+
+    def test_bad_structures(self):
+        test_structures_path = (
+            Path(__file__).resolve().parent.joinpath("test_bad_structures.json")
+        )
+        with open(test_structures_path, "r") as f:
+            bad_structures = json.load(f)
+        for doc in bad_structures:
+            doc["last_modified"] = doc["last_modified"]["$date"]
 
-from optimade.models.util import conlist
+        print("\n")
+        for ind, structure in enumerate(bad_structures):
+            with self.assertRaises(ManualValidationError):
+                try:
+                    _ = StructureResource(**StructureMapper.map_back(structure))
```

Comment:
Same as above, could it not just be?:
```suggestion
                    StructureResource(**StructureMapper.map_back(structure))
```